### PR TITLE
feat: Product Details Title Quantity 🔫

### PR DESCRIPTION
**Pull request by ziadali22**
- create xib, swift files view
- create two methods ( styling , check stock count and change style , pass model data to View )

### CODE REVIEW

**Review by claramounir — APPROVED** [review]

File: Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+struct ProductDetails {
+    var productTitle: String
+    var productCurrency: String
+    var productPrice: String
+    var productReviewAverage: Double
+    var productReviewCount: Int
+    var productStockCount: Int
+}
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift]: - A good approach is to add a view model as a nested type to the view. 
- Remove duplicates even in a properties names. 
- Immutable properties are preferred. 
- 
```suggestion
// MARK: - View Model
//
extension ProductTitleQuantityView { 
    struct ViewModel {
        let title: String
        let currency: String
        let price: String
        let reviewAverage: Double
        let reviewCount: Int
        let stockCount: Int
    }
}
```

File: Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
+//
+//  ProductTitleQuantityView.swift
+//  Dokan
+//
+//  Created by ziad on 21/07/2022.
+//
+
+import UIKit
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift:8]:
```suggestion
import UIKit

```

File: Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        } else if stockCount > 0 && stockCount <= 10 {
+            self.productStockBackgroundView.backgroundColor = .orange
+        } else {
+            self.productStockBackgroundView.backgroundColor = .red
+        }
+    }
+
+    // MARK: Configuration
+    
+    func passProductDetailsDataToView(productDetailsData: ProductDetails) {
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift]:
```suggestion
    func configureView(viewModel: ViewModel) {
```

File: Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    // MARK: Style
+
+    func stylingProductDetails() {
+        self.productStockBackgroundView.layer.cornerRadius = 10
+    }
+
+    func changeStockBackGroundColor(stockCount: Int) {
+        if stockCount > 10 {
+            self.productStockBackgroundView.backgroundColor = .green
+        } else if stockCount > 0 && stockCount <= 10 {
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift]: Already checked in the previous step
```suggestion
        } else if stockCount > 0 {
```

File: Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            self.productStockBackgroundView.backgroundColor = .orange
+        } else {
+            self.productStockBackgroundView.backgroundColor = .red
+        }
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift]:
```suggestion
        let color: UIColor
        if stockCount > 10 {
            color = .green
        } else if stockCount > 0 && stockCount <= 10 {
            color = .orange
        } else {
            color = .red
        }
        
        productStockBackgroundView.backgroundColor = color
```

File: Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        stylingProductDetails()
+    }
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        stylingProductDetails()
+    }
+    
+    // MARK: Style
+
+    func stylingProductDetails() {
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift]: - Keep an eye to the methods that should be private and other public/ internal methods. 
- Try to use extensions to decorate your code when possible.

File: Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.xib [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+            <point key="canvasLocation" x="137.68115942028987" y="109.82142857142857"/>
+        </view>
+    </objects>
+    <resources>
+        <image name="Star" width="12" height="12"/>
+        <systemColor name="systemBackgroundColor">
+            <color white="1" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
+        </systemColor>
+    </resources>
+</document>
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.xib]:
```suggestion
<?xml version="1.0" encoding="UTF-8"?>
<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="21208.1" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
    <device id="retina6_1" orientation="portrait" appearance="light"/>
    <dependencies>
        <deployment identifier="iOS"/>
        <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="21191"/>
        <capability name="System colors in document resources" minToolsVersion="11.0"/>
        <capability name="documents saved in the Xcode 8 format" minToolsVersion="8.0"/>
    </dependencies>
    <objects>
        <placeholder placeholderIdentifier="IBFilesOwner" id="-1" userLabel="File's Owner" customClass="ProductTitleQuantityView" customModule="Dokan" customModuleProvider="target">
            <connections>
                <outlet property="productCurrency" destination="379-1p-jyK" id="uP2-Nt-O5c"/>
                <outlet property="productPrice" destination="cjK-WF-LF9" id="RZ5-T1-ZDb"/>
                <outlet property="productReviewAverage" destination="htj-L6-46b" id="hqW-2I-SGj"/>
                <outlet property="productReviewCount" destination="aRY-cB-ww3" id="E32-2P-dVX"/>
                <outlet property="productStockBackgroundView" destination="XAZ-a0-IKP" id="pdm-1Q-gnL"/>
                <outlet property="productStockCount" destination="mwO-Wl-fdI" id="XII-9f-AgW"/>
                <outlet property="productTitle" destination="ntf-ut-53d" id="Tnh-nT-Ywx"/>
            </connections>
        </placeholder>
        <placeholder placeholderIdentifier="IBFirstResponder" id="-2" customClass="UIResponder"/>
        <view contentMode="scaleToFill" id="iN0-l3-epB">
            <rect key="frame" x="0.0" y="0.0" width="414" height="896"/>
            <autoresizingMask key="autoresizingMask" widthSizable="YES" heightSizable="YES"/>
            <subviews>
                <stackView opaque="NO" contentMode="scaleToFill" axis="vertical" spacing="10" translatesAutoresizingMaskIntoConstraints="NO" id="6Qm-o8-1Ol" userLabel="Main Components Stack">
                    <rect key="frame" x="0.0" y="0.0" width="414" height="896"/>
                    <subviews>
                        <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="TMA-2HD Wireless" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="ntf-ut-53d">
                            <rect key="frame" x="0.0" y="0.0" width="414" height="29"/>
                            <fontDescription key="fontDescription" type="boldSystem" pointSize="24"/>
                            <color key="textColor" red="0.047058823529411764" green="0.10196078431372549" blue="0.18823529411764706" alpha="1" colorSpace="calibratedRGB"/>
                            <nil key="highlightedColor"/>
                        </label>
                        <stackView opaque="NO" contentMode="scaleToFill" spacing="5" translatesAutoresizingMaskIntoConstraints="NO" id="G0F-Jq-KlI" userLabel="Price Stack">
                            <rect key="frame" x="0.0" y="39" width="414" height="19.5"/>
                            <subviews>
                                <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Rp." textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="379-1p-jyK">
                                    <rect key="frame" x="0.0" y="0.0" width="25" height="19.5"/>
                                    <fontDescription key="fontDescription" type="system" weight="medium" pointSize="16"/>
                                    <color key="textColor" red="0.99607843137254903" green="0.22745098039215686" blue="0.18823529411764706" alpha="1" colorSpace="calibratedRGB"/>
                                    <nil key="highlightedColor"/>
                                </label>
                                <label opaque="NO" userInteractionEnabled="NO" contentMode="left" verticalHuggingPriority="251" text="1.500.000" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="cjK-WF-LF9">
                                    <rect key="frame" x="30" y="0.0" width="384" height="19.5"/>
                                    <fontDescription key="fontDescription" type="system" weight="medium" pointSize="16"/>
                                    <color key="textColor" red="0.99607843137254903" green="0.22745098039215686" blue="0.18823529411764706" alpha="1" colorSpace="calibratedRGB"/>
                                    <nil key="highlightedColor"/>
                                </label>
                            </subviews>
                        </stackView>
                        <stackView opaque="NO" contentMode="scaleToFill" distribution="equalSpacing" alignment="center" translatesAutoresizingMaskIntoConstraints="NO" id="Wte-IS-ayG">
                            <rect key="frame" x="0.0" y="68.5" width="414" height="827.5"/>
                            <subviews>
                                <stackView opaque="NO" contentMode="scaleToFill" distribution="fillProportionally" spacing="10" translatesAutoresizingMaskIntoConstraints="NO" id="xdP-Xh-LIc" userLabel="Review Stack">
                                    <rect key="frame" x="0.0" y="406.5" width="123" height="15"/>
                                    <subviews>
                                        <stackView opaque="NO" contentMode="scaleToFill" spacing="4" translatesAutoresizingMaskIntoConstraints="NO" id="qWN-ON-yjc" userLabel="Review Average Stack">
                                            <rect key="frame" x="0.0" y="0.0" width="40" height="15"/>
                                            <subviews>
                                                <imageView clipsSubviews="YES" userInteractionEnabled="NO" contentMode="scaleAspectFill" horizontalHuggingPriority="251" verticalHuggingPriority="251" image="Star" translatesAutoresizingMaskIntoConstraints="NO" id="8iz-ZD-oKg">
                                                    <rect key="frame" x="0.0" y="0.0" width="15" height="15"/>
                                                    <constraints>
                                                        <constraint firstAttribute="width" constant="15" id="MaY-cU-nqh"/>
                                                        <constraint firstAttribute="height" constant="15" id="qHP-yN-IBa"/>
                                                    </constraints>
                                                </imageView>
                                                <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="4.6" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="htj-L6-46b">
                                                    <rect key="frame" x="19" y="0.0" width="21" height="15"/>
                                                    <fontDescription key="fontDescription" type="system" pointSize="14"/>
                                                    <color key="textColor" red="0.047058823529411764" green="0.10196078431372549" blue="0.18823529411764706" alpha="1" colorSpace="calibratedRGB"/>
                                                    <nil key="highlightedColor"/>
                                                </label>
                                            </subviews>
                                        </stackView>
                                        <stackView opaque="NO" contentMode="scaleToFill" spacing="2" translatesAutoresizingMaskIntoConstraints="NO" id="0zM-AV-oNl" userLabel="Reviews Count Stack">
                                            <rect key="frame" x="50" y="0.0" width="73" height="15"/>
                                            <subviews>
                                                <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="252" verticalHuggingPriority="251" text="86" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="aRY-cB-ww3">
                                                    <rect key="frame" x="0.0" y="0.0" width="18" height="15"/>
                                                    <fontDescription key="fontDescription" type="system" pointSize="14"/>
                                                    <color key="textColor" red="0.047058823529411764" green="0.10196078431372549" blue="0.18823529411764706" alpha="1" colorSpace="calibratedRGB"/>
                                                    <nil key="highlightedColor"/>
                                                </label>
                                                <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Reviews" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="UZO-Ah-tmN">
                                                    <rect key="frame" x="20" y="0.0" width="53" height="15"/>
                                                    <fontDescription key="fontDescription" type="system" pointSize="14"/>
                                                    <color key="textColor" red="0.047058823529411764" green="0.10196078431372549" blue="0.18823529411764706" alpha="1" colorSpace="calibratedRGB"/>
                                                    <nil key="highlightedColor"/>
                                                </label>
                                            </subviews>
                                        </stackView>
                                    </subviews>
                                </stackView>
                                <stackView opaque="NO" contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="nGy-aF-yWA" userLabel="Stock Count Stack">
                                    <rect key="frame" x="312" y="404" width="102" height="20"/>
                                    <subviews>
                                        <view contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="XAZ-a0-IKP" userLabel="BackGround View">
                                            <rect key="frame" x="0.0" y="0.0" width="102" height="20"/>
                                            <subviews>
                                                <stackView opaque="NO" contentMode="scaleToFill" translatesAutoresizingMaskIntoConstraints="NO" id="bPg-6f-fHR">
                                                    <rect key="frame" x="8" y="0.0" width="86" height="20"/>
                                                    <subviews>
                                                        <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Tersedia :" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="pNQ-6a-LSv">
                                                            <rect key="frame" x="0.0" y="0.0" width="63" height="20"/>
                                                            <fontDescription key="fontDescription" type="system" weight="medium" pointSize="12"/>
                                                            <color key="textColor" red="0.22745098039215686" green="0.60784313725490191" blue="0.47843137254901957" alpha="1" colorSpace="calibratedRGB"/>
                                                            <nil key="highlightedColor"/>
                                                        </label>
                                                        <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="250" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="mwO-Wl-fdI">
                                                            <rect key="frame" x="63" y="0.0" width="23" height="20"/>
                                                            <fontDescription key="fontDescription" type="system" weight="medium" pointSize="12"/>
                                                            <color key="textColor" red="0.2274509804" green="0.60784313729999995" blue="0.47843137250000001" alpha="1" colorSpace="calibratedRGB"/>
                                                            <nil key="highlightedColor"/>
                                                        </label>
                                                    </subviews>
                                                </stackView>
                                            </subviews>
                                            <color key="backgroundColor" red="0.93333333333333335" green="0.98039215686274506" blue="0.96470588235294119" alpha="1" colorSpace="calibratedRGB"/>
                                            <constraints>
                                                <constraint firstItem="bPg-6f-fHR" firstAttribute="top" secondItem="XAZ-a0-IKP" secondAttribute="top" id="RcJ-Ij-PLH"/>
                                                <constraint firstAttribute="bottom" secondItem="bPg-6f-fHR" secondAttribute="bottom" id="WB3-ez-PlR"/>
                                                <constraint firstItem="bPg-6f-fHR" firstAttribute="trailing" secondItem="XAZ-a0-IKP" secondAttribute="trailingMargin" id="fmg-aF-SAW"/>
                                                <constraint firstItem="bPg-6f-fHR" firstAttribute="centerX" secondItem="XAZ-a0-IKP" secondAttribute="centerX" id="xgw-yJ-oUm"/>
                                            </constraints>
                                        </view>
                                    </subviews>
                                    <constraints>
                                        <constraint firstAttribute="height" constant="20" id="Ma8-G2-L1E"/>
                                        <constraint firstAttribute="bottom" secondItem="XAZ-a0-IKP" secondAttribute="bottom" id="abh-84-hIl"/>
                                        <constraint firstAttribute="trailing" secondItem="XAZ-a0-IKP" secondAttribute="trailing" id="ibm-NB-Nzy"/>
                                        <constraint firstItem="XAZ-a0-IKP" firstAttribute="leading" secondItem="nGy-aF-yWA" secondAttribute="leading" id="mlc-mL-o7r"/>
                                        <constraint firstItem="XAZ-a0-IKP" firstAttribute="top" secondItem="nGy-aF-yWA" secondAttribute="top" id="qSB-Dh-MEm"/>
                                    </constraints>
                                </stackView>
                            </subviews>
                        </stackView>
                    </subviews>
                </stackView>
            </subviews>
            <color key="backgroundColor" systemColor="systemBackgroundColor"/>
            <constraints>
                <constraint firstItem="6Qm-o8-1Ol" firstAttribute="leading" secondItem="iN0-l3-epB" secondAttribute="leading" id="R6c-XZ-lQW"/>
                <constraint firstAttribute="trailing" secondItem="6Qm-o8-1Ol" secondAttribute="trailing" id="TwE-Nx-SJt"/>
                <constraint firstAttribute="bottom" secondItem="6Qm-o8-1Ol" secondAttribute="bottom" id="jL1-sy-WYc"/>
                <constraint firstItem="6Qm-o8-1Ol" firstAttribute="top" secondItem="iN0-l3-epB" secondAttribute="top" id="tKk-EQ-D90"/>
            </constraints>
            <point key="canvasLocation" x="43" y="-905"/>
        </view>
    </objects>
    <resources>
        <image name="Star" width="12" height="12"/>
        <systemColor name="systemBackgroundColor">
            <color white="1" alpha="1" colorSpace="custom" customColorSpace="genericGamma22GrayColorSpace"/>
        </systemColor>
    </resources>
</document>
```

Give this a try
- Use stack views completely. 
- Spaces multiples of 4

File: Dokan/Dokan.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -110,6 +114,25 @@
 /* End PBXFrameworksBuildPhase section */
 
 /* Begin PBXGroup section */
+		273F1DE32874A4A4006485D6 /* Registratio9nTest */ = {
```
Comment (ahmdmhasn) [inline Dokan/Dokan.xcodeproj/project.pbxproj]: Typo `RegistrationTest`

**Review by ahmdmhasn — COMMENTED** [review]
🚀

File: Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    // MARK: Style
+
+    private func stylingProductDetails() {
+        self.productStockBackgroundView.layer.cornerRadius = 10
+    }
+
+    func changeStockBackGroundColor(stockCount: Int) {
+        
+        let color: UIColor
+
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift]:
```suggestion
        let color: UIColor
```

File: Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+        stylingProductDetails()
+    }
+    
+    // MARK: Style
+
+    private func stylingProductDetails() {
+        self.productStockBackgroundView.layer.cornerRadius = 10
+    }
+
+    func changeStockBackGroundColor(stockCount: Int) {
```
Comment (ahmdmhasn) [inline Dokan/Dokan/Scenes/ProductDetails/TitleQuantity/ProductTitleQuantityView.swift]: It should be private and added to configure method.

**Review by ahmdmhasn — APPROVED** [review]
Neat and clean. 🎆

**Review by RaniaZeid — APPROVED** [review]

**Review by oradwanomar — APPROVED** [review]

**Pull request merged**